stubservice/main.go

```diff
@@ -34,6 +34,8 @@ const (
 
 var (
 	baseURL = os.Getenv("BASE_URL")
+	
+	baseBouncerURL = os.Getenv("BOUNCER_URL")
```

Comment:
Can we rename the env var to `BOUNCER_BASE_URL` for consistency? Though, we'd need the local var to be `bouncerBaseURL` as well, which I think is fine too. If that is fine with you, then here is a diff to apply with all the changes so you don't have to do everything:

```diff
diff --git a/stubservice/main.go b/stubservice/main.go
index 4f7a0a45..fa6eed7e 100644
--- a/stubservice/main.go
+++ b/stubservice/main.go
@@ -34,8 +34,8 @@ const (
 
 var (
 	baseURL = os.Getenv("BASE_URL")
-	
-	baseBouncerURL = os.Getenv("BOUNCER_URL")
+
+	bouncerBaseURL = os.Getenv("BOUNCER_BASE_URL")
 
 	hmacKey        = os.Getenv("HMAC_KEY")
 	hmacTimeoutEnv = os.Getenv("HMAC_TIMEOUT")
@@ -71,8 +71,8 @@ func init() {
 		logrus.Fatal("BASE_URL is required")
 	}
 
-	if baseBouncerURL == "" {
-		baseBouncerURL = "https://download.mozilla.org/"
+	if bouncerBaseURL == "" {
+		bouncerBaseURL = "https://download.mozilla.org/"
 	}
 
 	switch returnMode {
@@ -199,19 +199,19 @@ func main() {
 			}
 
 			store := backends.NewGCS(gcsStorageClient, gcsBucket, time.Hour*24)
-			stubHandler = stubhandlers.NewRedirectHandler(store, cdnPrefix, gcsPrefix, baseBouncerURL)
+			stubHandler = stubhandlers.NewRedirectHandler(store, cdnPrefix, gcsPrefix, bouncerBaseURL)
 		} else {
 			logrus.WithField("backend", storageBackend).Fatal("Unsupported storage backend")
 		}
 	} else {
 		logrus.Info("Starting in direct mode")
-		stubHandler = stubhandlers.NewDirectHandler(baseBouncerURL)
+		stubHandler = stubhandlers.NewDirectHandler(bouncerBaseURL)
 	}
 
 	stubService := stubhandlers.NewStubService(
 		stubHandler,
 		attributioncode.NewValidator(hmacKey, hmacTimeout),
-		baseBouncerURL,
+		bouncerBaseURL,
 	)
 
 	mux := http.NewServeMux()
diff --git a/stubservice/stubhandlers/direct.go b/stubservice/stubhandlers/direct.go
index 10f3a6be..e1d441b2 100644
--- a/stubservice/stubhandlers/direct.go
+++ b/stubservice/stubhandlers/direct.go
@@ -13,14 +13,14 @@ import (
 type directHandler struct {
 	sfGroup *singleflight.Group
 
-	BaseBouncerURL string
+	BouncerBaseURL string
 }
 
 // NewDirectHandler returns a new direct type handler
-func NewDirectHandler(baseBouncerURL string) StubHandler {
+func NewDirectHandler(bouncerBaseURL string) StubHandler {
 	return &directHandler{
-		sfGroup: new(singleflight.Group),
-		BaseBouncerURL: baseBouncerURL,
+		sfGroup:        new(singleflight.Group),
+		BouncerBaseURL: bouncerBaseURL,
 	}
 }
 
@@ -32,7 +32,7 @@ func (s *directHandler) ServeStub(w http.ResponseWriter, req *http.Request, code
 	os := query.Get("os")
 	attributionCode := code.URLEncode()
 
-	stub, err := sfFetchStub(s.sfGroup, bouncerURL(product, lang, os, s.BaseBouncerURL))
+	stub, err := sfFetchStub(s.sfGroup, bouncerURL(product, lang, os, s.BouncerBaseURL))
 	if err != nil {
 		return errors.Wrap(err, "fetchStub")
 	}
diff --git a/stubservice/stubhandlers/redirect.go b/stubservice/stubhandlers/redirect.go
index 9af9f75e..b4f544d3 100644
--- a/stubservice/stubhandlers/redirect.go
+++ b/stubservice/stubhandlers/redirect.go
@@ -31,11 +31,11 @@ type redirectHandler struct {
 
 	sfGroup *singleflight.Group
 
-	BaseBouncerURL string
+	BouncerBaseURL string
 }
 
 // NewRedirectHandler returns a new StubHandler
-func NewRedirectHandler(storage backends.Storage, cdnPrefix, keyPrefix string, baseBouncerURL string) StubHandler {
+func NewRedirectHandler(storage backends.Storage, cdnPrefix, keyPrefix string, bouncerBaseURL string) StubHandler {
 	return &redirectHandler{
 		CDNPrefix: cdnPrefix,
 		KeyPrefix: keyPrefix,
@@ -44,7 +44,7 @@ func NewRedirectHandler(storage backends.Storage, cdnPrefix, keyPrefix string, b
 
 		sfGroup: new(singleflight.Group),
 
-		BaseBouncerURL: baseBouncerURL,
+		BouncerBaseURL: bouncerBaseURL,
 	}
 }
 
@@ -56,7 +56,7 @@ func (s *redirectHandler) ServeStub(w http.ResponseWriter, req *http.Request, co
 	os := query.Get("os")
 	attributionCode := code.URLEncode()
 
-	bURL := bouncerURL(product, lang, os, s.BaseBouncerURL)
+	bURL := bouncerURL(product, lang, os, s.BouncerBaseURL)
 
 	cdnURL, err := redirectResponse(bURL)
 	if err != nil {
diff --git a/stubservice/stubhandlers/stubhandler_test.go b/stubservice/stubhandlers/stubhandler_test.go
index d94fe29b..eb5e834d 100644
--- a/stubservice/stubhandlers/stubhandler_test.go
+++ b/stubservice/stubhandlers/stubhandler_test.go
@@ -368,11 +368,11 @@ func TestDirectFull(t *testing.T) {
 }
 
 func TestStubServiceErrorCases(t *testing.T) {
-	baseBouncerURL := "https://download.mozilla.org/"
+	bouncerBaseURL := "https://download.mozilla.org/"
 	svc := NewStubService(
-		NewDirectHandler(baseBouncerURL),
+		NewDirectHandler(bouncerBaseURL),
 		&attributioncode.Validator{},
-		baseBouncerURL,
+		bouncerBaseURL,
 	)
 
 	fetchURL := func(url string) *httptest.ResponseRecorder {
diff --git a/stubservice/stubhandlers/stubservice.go b/stubservice/stubhandlers/stubservice.go
index 839db891..1da2382f 100644
--- a/stubservice/stubhandlers/stubservice.go
+++ b/stubservice/stubhandlers/stubservice.go
@@ -14,14 +14,14 @@ type stubService struct {
 
 	AttributionCodeValidator *attributioncode.Validator
 
-	BaseBouncerURL string
+	BouncerBaseURL string
 }
 
-func NewStubService(stubHandler StubHandler, validator *attributioncode.Validator, baseBouncerUrl string) http.Handler {
+func NewStubService(stubHandler StubHandler, validator *attributioncode.Validator, bouncerBaseURL string) http.Handler {
 	return &stubService{
 		Handler:                  stubHandler,
 		AttributionCodeValidator: validator,
-		BaseBouncerURL: baseBouncerUrl,
+		BouncerBaseURL:           bouncerBaseURL,
 	}
 }
 
@@ -32,7 +32,7 @@ func (s *stubService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	query := req.URL.Query()
 
 	redirectBouncer := func() {
-		backupURL := bouncerURL(query.Get("product"), query.Get("lang"), query.Get("os"), s.BaseBouncerURL)
+		backupURL := bouncerURL(query.Get("product"), query.Get("lang"), query.Get("os"), s.BouncerBaseURL)
 		http.Redirect(w, req, backupURL, http.StatusFound)
 	}
 
diff --git a/stubservice/stubhandlers/utils.go b/stubservice/stubhandlers/utils.go
index fca26342..0879746d 100644
--- a/stubservice/stubhandlers/utils.go
+++ b/stubservice/stubhandlers/utils.go
@@ -20,12 +20,12 @@ var stubClient = &http.Client{
 	Timeout: 30 * time.Second,
 }
 
-func bouncerURL(product, lang, os string, baseBouncerUrl string) string {
+func bouncerURL(product, lang, os, baseURL string) string {
 	v := url.Values{}
 	v.Set("product", product)
 	v.Set("lang", lang)
 	v.Set("os", os)
-	return baseBouncerUrl + "?" + v.Encode()
+	return baseURL + "?" + v.Encode()
 }
 
 type modifiedStub struct {
```

Reply:
I guess we could update the README file too:

```diff
diff --git a/stubservice/README.md b/stubservice/README.md
index 4d9bf977..52644d7a 100644
--- a/stubservice/README.md
+++ b/stubservice/README.md
@@ -25,9 +25,9 @@ format](https://golang.org/pkg/time/#ParseDuration).
 
 If set, tracebacks will be sent to [Sentry](https://getsentry.com/).
 
-### BOUNCER_URL
+### BOUNCER_BASE_URL
 
-Bouncer root URL. The default value is: `https://download.mozilla.org/`.
+Bouncer base URL. The default value is: `https://download.mozilla.org/`.
 
 ### RETURN_MODE
 
```